syntactic-paths/src/main/java/com/palantir/util/syntacticpath/Path.java

```diff
@@ -330,13 +322,52 @@ private String toStringInternal() {
             String prefix = isAbsolute ? SEPARATOR : "";
             String suffix = isFolder ? SEPARATOR : "";
 
-            return prefix + PATH_JOINER.join(segments) + suffix;
+            return prefix + join(segments) + suffix;
         }
     }
 
     /** Returns the string representation of this (non-normalized) path. */
     @Override
     public String toString() {
-        return stringRepresentation.get();
+        return stringRepresentation;
+    }
+
+    static String join(List<String> segments) {
```

Comment:
I'd use whatever makes the calling code most intuitive/legible. for instance, I would prefer one of
````
segments = split(path, SEPARATOR);
segments = splitBySeparator(path);
segments = BY_SEPARATOR.split(path);
````
over
````
segments = split(path);
segments = SPLITTER.split(path);
````